### PR TITLE
Feature/address select contact person configurable

### PR DESCRIPTION
**Pull request by cjohannsen**
[IO pullrequest](https://github.com/plentymarkets/plugin-io/pull/436)

### All changes meet the following requirements
- [x] Changelog entry was added
- [x] Changes have been tested
- [x] Plugin can be built

@plentymarkets/ceres-io

### CODE REVIEW

File: resources/lang/de/Config.properties [hunk 1]
```diff
@@ -319,26 +319,28 @@ billingAddressEnShowLabel = Rechnungsadressfelder im Adressformular anzeigen (UK
...
-addressFieldEnTitle = Titel
-addressFieldEnBirthday = Geburtsdatum
-addressFieldEnName4 = Namenszusatz
+addressFieldBirthday = Geburtsdatum
+addressFieldContactPerson = Kontaktperson
 addressFieldEnAddress2 = Adresse 2
 addressFieldEnAddress3 = Adresse 3
 addressFieldEnAddress4 = Adresse 4
+addressFieldEnBirthday = Geburtsdatum
+addressFieldEnContactPerson = Kontaktperson
```
Comment (cjohannsen) [inline resources/lang/de/Config.properties:330]: @fmutschler addressFieldEnContactPerson

File: resources/lang/de/Widget.properties [hunk 2]
```diff
@@ -383,26 +383,28 @@ addressFieldsShippingDE = "Lieferadressfelder anzeigen (DE)"
...
-addressFieldStateId = Bundesland
-addressFieldPhoneNumber = Telefonnummer
-addressFieldEnName1 = Firma
-addressFieldEnVatNumber = Ust.-Nr.
-addressFieldEnSalutation = Anrede
-addressFieldEnTitle = Titel
-addressFieldEnBirthday = Geburtsdatum
-addressFieldEnName4 = Namenszusatz
+addressFieldBirthday = Geburtsdatum
+addressFieldContactPerson = Kontaktperson
```
Comment (cjohannsen) [inline resources/lang/de/Widget.properties:389]: @fmutschler addressFieldContactPerson

File: resources/lang/en/Config.properties [hunk 3]
```diff
@@ -319,26 +319,28 @@ billingAddressEnShowLabel = Show invoice address fields in address form (UK)
...
-addressFieldStateId = State
-addressFieldPhoneNumber = Phone number
-addressFieldEnName1 = Company
-addressFieldEnVatNumber = VAT number
-addressFieldEnSalutation = Form of address
-addressFieldEnTitle = Title
-addressFieldEnBirthday = Date of birth
-addressFieldEnName4 = Name affix
+addressFieldBirthday = Date of birth
+addressFieldContactPerson = Contact person
```
Comment (cjohannsen) [inline resources/lang/en/Config.properties:325]: @fmutschler addressFieldContactPerson

File: resources/lang/de/Config.properties [hunk 4]
```diff
@@ -319,26 +319,28 @@ billingAddressEnShowLabel = Rechnungsadressfelder im Adressformular anzeigen (UK
...
-addressFieldStateId = Bundesland
-addressFieldPhoneNumber = Telefonnummer
-addressFieldEnName1 = Firma
-addressFieldEnVatNumber = USt.-Nr.
-addressFieldEnSalutation = Anrede
-addressFieldEnTitle = Titel
-addressFieldEnBirthday = Geburtsdatum
-addressFieldEnName4 = Namenszusatz
+addressFieldBirthday = Geburtsdatum
+addressFieldContactPerson = Kontaktperson
```
Comment (cjohannsen) [inline resources/lang/de/Config.properties:325]: @fmutschler addressFieldContactPerson

File: resources/lang/de/Widget.properties [hunk 5]
```diff
@@ -383,26 +383,28 @@ addressFieldsShippingDE = "Lieferadressfelder anzeigen (DE)"
...
 addressFieldEnAddress2 = Adresse 2
 addressFieldEnAddress3 = Adresse 3
 addressFieldEnAddress4 = Adresse 4
+addressFieldEnBirthday = Geburtsdatum
+addressFieldEnContactPerson = Kontaktperson
```
Comment (cjohannsen) [inline resources/lang/de/Widget.properties:394]: @fmutschler addressFieldEnContactPerson

File: resources/lang/en/Config.properties [hunk 6]
```diff
@@ -319,26 +319,28 @@ billingAddressEnShowLabel = Show invoice address fields in address form (UK)
...
 addressFieldEnAddress2 = Address 2
 addressFieldEnAddress3 = Address 3
 addressFieldEnAddress4 = Address 4
+addressFieldEnBirthday = Date of birth
+addressFieldEnContactPerson = Contact person
```
Comment (cjohannsen) [inline resources/lang/en/Config.properties:330]: @fmutschler addressFieldEnContactPerson

File: resources/lang/en/Widget.properties [hunk 7]
```diff
@@ -383,26 +383,28 @@ addressFieldsShippingDE = "Show shipping address fields (DE)"
...
-addressFieldStateId = State
-addressFieldPhoneNumber = Phone number
-addressFieldEnName1 = Company
-addressFieldEnVatNumber = VAT number
-addressFieldEnSalutation = Form of address
-addressFieldEnTitle = Title
-addressFieldEnBirthday = Date of birth
-addressFieldEnName4 = Name affix
+addressFieldBirthday = Date of birth
+addressFieldContactPerson = Contact person
```
Comment (cjohannsen) [inline resources/lang/en/Widget.properties:389]: @fmutschler addressFieldContactPerson

File: resources/lang/en/Widget.properties [hunk 8]
```diff
@@ -383,26 +383,28 @@ addressFieldsShippingDE = "Show shipping address fields (DE)"
...
 addressFieldEnAddress2 = Address 2
 addressFieldEnAddress3 = Address 3
 addressFieldEnAddress4 = Address 4
+addressFieldEnBirthday = Date of birth
+addressFieldEnContactPerson = Contact person
```
Comment (cjohannsen) [inline resources/lang/en/Widget.properties:394]: @fmutschler addressFieldEnContactPerson

File: resources/js/src/app/components/customer/AddressInputGroup.js [hunk 9]
```diff
@@ -148,6 +148,24 @@ Vue.component("address-input-group", {
             const isRequired = this.isInRequiredFields(locale, addressKey);
 
             return translation + (isRequired ? "*" : "");
+        },
+
+        areNameFieldsShown(locale, keyPrefix)
+        {
+            const condition1 = this.isInOptionalFields(locale, `${keyPrefix}.salutation`) && this.isInOptionalFields(locale, `${keyPrefix}.contactPerson`) && this.value.addressSalutation === 2;
```
Comment (lks-mtzn) [inline resources/js/src/app/components/customer/AddressInputGroup.js]: Kannst du diese conditions noch sinnvoll benennen?

**Review by lks-mtzn — APPROVED** [review]

**Pull request merged**